internal/monitors/cadvisor/kubeletstats.go

```diff
@@ -153,3 +171,22 @@ func (kip *kubeletInfoProvider) getAllContainersLatestStats() ([]info.ContainerI
 	}
 	return result, nil
 }
+
+func (kip *kubeletInfoProvider) GetEphemeralStatsFromPods() ([]stats.PodStats, error) {
+	if !kip.includePodEphemeralStats {
```

Comment:
Do the conditional check in the function that calls this, not in here.  Also use `output.HasEnabledMetricInGroup`, not a config flag on the monitor.

Reply:
Since the caller of this method is common to both the `kubelet-stats` and `cadvisor` monitors, would it not be better to do this check here? This check is only valid for `kubelet-stat` monitor. `cadvisor` monitor does not have these metrics.

Reply:
No, this function says it "gets the ephemeral stats from pods" and it should not lie.

Reply:
Added a `GroupsEnablerHelper` to keep track of enabled groups in the converter. Let me know what you think.